### PR TITLE
Tas 151 make auth token editable

### PR DESCRIPTION
**Pull request by csrubin**
# Impact

This PR allows for event passwords to be optional and editable. Additionally adds optional email input, and sends event details upon save. Remove token dialog flow.

### CODE REVIEW

File: how2meet/routers/events.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def create_event(event: schemas.EventCreate, db: Session = Depends(get_db)) -> m
 
     Args:
         event: The event to be created.
-        db: The database session. Defaults to Depends(get_db).
+        db: The database session. Defaults to Depends(geft_db).
```
Comment (BenMemberg) [inline how2meet/routers/events.py:26]: I had assumed you were adding this whole sentence or something... but this was just a straight up stray "f" 😂

File: how2meet/ui/components/event_editor.py [hunk 2]
```diff
@@ -179,43 +204,54 @@ async def save(self, on_save=None):
...
             ui.notify(f"Sorry we ran into an error trying to save your event!")
             return
 
-        if status.is_success:
+        if response.is_success:
             ui.notify("Saved!")
-            token_dialog = TokenDialog(model_dump["auth_token"])
-            await token_dialog.render(on_save)
+
+            logger.info("Response:")
```
Comment (BenMemberg) [inline how2meet/ui/components/event_editor.py:240]: Probably remove these debugging loggers

File: how2meet/ui/components/event_editor.py [hunk 3]
```diff
@@ -179,43 +204,54 @@ async def save(self, on_save=None):
...
+            logger.info(response)
+            logger.info("Response Content:")
+            logger.info(response.content)
+            api_data = json.loads(response.content.decode("utf-8"))
+
+            email = api_data["email"]  # TODO: access like this or through class interface?
```
Comment (BenMemberg) [inline how2meet/ui/components/event_editor.py:246]: What does this comment mean?

File: how2meet/ui/components/event_editor.py [hunk 4]
```diff
@@ -179,43 +204,54 @@ async def save(self, on_save=None):
...
+            if email:
+                try:
+                    send_email(email, f"How2Meet Event Details: {api_data['name']}", str(model_data))
+                except smtplib.SMTPException as e:
+                    ui.notify(f"Failed to send email with error: {e}")
+
+            ui.open(URL_EVENT_HOME.format(event_id=self.event_id))
```
Comment (BenMemberg) [inline how2meet/ui/components/event_editor.py:253]: I like the redirect 👍

**Review by BenMemberg — APPROVED** [review]
Just a couple small things 👍

File: how2meet/ui/components/event_editor.py (hunk 3 above)
Comment (csrubin) [reply how2meet/ui/components/event_editor.py]: Remnant from when we were trying to figure out the response content thing

**Pull request merged**